desktop/core/src/desktop/lib/fsmanager.py

```diff
@@ -117,8 +117,13 @@ def _get_client_cached(fs, name, user):
   global CLIENT_CACHE
   if CLIENT_CACHE is None:
     CLIENT_CACHE = {}
-  # We don't want to cache by username when IDBroker not enabled
-  cache_key = _get_cache_key(fs, name, user) if conf_idbroker.is_idbroker_enabled(fs) else _get_cache_key(fs, name)
+
+  # We don't want to cache by username when IDBroker or RAZ is not enabled
+  if (conf_idbroker.is_idbroker_enabled(fs) or RAZ.IS_ENABLED.get()):
+    cache_key = _get_cache_key(fs, name, user)
+  else:
+    cache_key = _get_cache_key(fs, name)
```

Comment:
Why are we not caching by username in the default case? That would be a good candidate for the comment.

Reply:
We are not caching per user basis in the default case, because in the default case there is no user mapping scenario like in IDBroker or RAZ.

For FS like S3 and ABFS: Default case is via access key and secret which just allows everyone to access everything.

For FS like HDFS and upcoming Ozone: This user mapping stuff is handled behind the scenes and we are doing impersonation for it.

Reply:
Added as comment.

Reply:
Awesome!